Add api base setter for changing the base url for nomics api calls

spencewood opened the pull request:
This adds a `NOMICS_API_BASE` public property setter to the instance of the Nomics API client. Changing the base url for all api calls for a ticker would go something like this:

```js
const nomics = new Nomics({ apiKey: Key() });
nomics.NOMICS_API_BASE = "https://new-api.nomics.com";
const currencies = await nomics.currenciesTicker({ interval: ["1d"] });
```

I'm up for changing the property name or making is more obscure, but the type mappings are included when this library is built, so it will show up in an IDE for anyone using this with Typescript.

spencewood commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Add way to change base url for JavaScript client](https://trello.com/c/c7XumaHv/420-add-way-to-change-base-url-for-javascript-client)

trestrantham commented on src/api/currencies_ticker.ts:
```diff
@@ -54,7 +55,9 @@ const currenciesTicker = async (
 
-  return fetchJSON(`${CURRENCIES_TICKER_URL}?${objToUrlParams(objParams)}`);
+  return fetchJSON(
+    `${apiBase}${CURRENCIES_TICKER_PATH}?${objToUrlParams(objParams)}`
```
Can this be a getter of `NOMICS_API_BASE` from `index.ts` rather than passing the URL around?

spencewood replied:
Yeah I agree. I didn't like this much either. The only other way that comes to mind is to turn this variable into a static property and then reference it that way inside the ticker functions instead of getting it via parameter.

spencewood commented:
@trestrantham Ok, I've updated this to use a static property instead, which I like a lot more. The extra benefit of this way is that you can change the base url at any time, including before or after you instantiate the Nomics class.

Updated usage:

```
Nomics.NOMICS_API_BASE = "https://new-api.nomics.com";
const nomics = new Nomics({ apiKey: Key() });
const currencies = await nomics.currenciesTicker({ interval: ["1d"] });
```

trestrantham commented:
Very nice! That will be super helpful with testing too. 😎
